### ISSUE TITLE
[Feature] Specifying select fields in .yml for snapshots

### ISSUE THREAD

**Issue by asarraf** [dbt-labs/dbt-core#11180]
### Is this your first time submitting a feature request?

- [X] I have read the [expectations for open source contributors](https://docs.getdbt.com/docs/contributing/oss-expectations)
- [X] I have searched the existing issues, and I could not find an existing issue for this feature
- [X] I am requesting a straightforward extension of existing dbt functionality, rather than a Big Idea better suited to a discussion

### Describe the feature

With the new way of creating and maintaing Snapshots, I see that there isn't a way to specify the list of columns that we want want to provide.
This issue is to add a functionality to accept `select` as one of the items in the yml that can be used to do a 
`select col1, col2.. from model`

instead of creating a ephemeral model and then doing `select *`

### Who will this benefit?

This will help whoever is using the DBT latest version so that they don't have to maintain an ephemeral models as well in the repository.

### Are you interested in contributing this feature?

Yes, I am stilll learning.

### Anything else?

References -> https://docs.getdbt.com/docs/build/snapshots#configuring-snapshots

**Comment by jeremyyeo** [comment]
The recommendation is to use an ephemeral (where you can specify the columns you need from your source) and snap that instead 😊

**Comment by asarraf** [comment]
Yes, I am suggesting if we want to use a materialized table with only a specific set of columns w/o any ephemeral model, the `--select` could be quite beneficial. Why create one more model, if we can add select columns. Seems more efficient.

**Comment by graciegoheen** [comment]
The benefit of storing this "select" logic in an ephemeral model is that you can then `test` / `show` that model before snapshotting it. I'm going to close this issue out for now, but we can always reopen if there's a lot of interest in this feature!

**Comment by asarraf** [comment]
> The benefit of storing this "select" logic in an ephemeral model is that you can then test/show that model before snapshotting it. I'm going to close this issue out for now, but we can always reopen if there's a lot of interest in this feature!

The suggestion is not to stop supporting ephemeral models. I am just suggesting to complement it with selectt.
